packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts

```diff
@@ -124,19 +146,72 @@ export class SampledTelemetryHelper implements IDisposable {
 		return returnValue;
 	}
 
+	/**
+	 * Accumulates the custom data and sends it to the logger every {@link SampledTelemetryHelper.sampleThreshold} calls.
+	 *
+	 * @param customData -
+	 * A record storing the custom data to be accumulated and eventually logged.
+	 */
+	private accumulateCustomData(customData: Record<TCustomMetrics, number>): void {
+		for (const key of Object.keys(customData) as TCustomMetrics[]) {
+			this.dataSums[key] = (this.dataSums[key] ?? 0) + customData[key];
+			this.dataMaxes[key] = Math.max(
+				this.dataMaxes[key] ?? Number.NEGATIVE_INFINITY,
+				customData[key],
+			);
+		}
+	}
+
+	/**
+	 * Computes average and sets new names for keys.
+	 *
+	 * @param customData - A record storing the custom data that has been accumulated over {@link SampledTelemetryHelper.sampleThreshold} times.
+	 * @param counts - The number of times the {@link SampledTelemetryHelper.measure} has been called.
+	 * @returns A record with the average and maximum values of the custom data.
+	 */
+	private processCustomData(
+		customData: Record<TCustomMetrics, number>,
+		counts: number,
+	): Record<string, number> {
+		const processedCustomData: Record<string, number> = {};
+
+		for (const key of Object.keys(customData) as TCustomMetrics[]) {
+			if (this.dataSums[key] !== undefined) {
+				processedCustomData[`avg${key}`] = roundToDecimalPlaces(
```

Comment:
Nit: I would recommend "average" over "avg". We should generally avoid abbreviations, as they represent a potential accessibility issue.

Reply:
This becomes part of a network payload though. It's not a super-high-frequency event, but we do expect to get them from every client, so I think in this case I wouldn't instinctively go for the expanded version.

Reply:
And even when they are _**really**_ common, it doesn't hurt to be more verbose IMO

Reply:
I guess I have a hard time believing expanding "avg" to "average" in these payloads would have a significant impact. But I won't block on it. "avg" is at least common enough that I'm not super concerned about it.